### DISCUSSION TITLE
Modify parameters for an already loaded scene

### DISCUSSION THREAD

**Discussion by michiruf** [FirstGearGames/FishNet#465] [Q&A - DO NOT USE - ASK ON DISCORD]
Hi there!

Im currently implementing a multi scene game with (hopefully at some point) a dedicated server.
To connect the players, I have an Entrypoint scene on the server and clients connecting to this create their player and get moved to their specific scenes they can play in then.

My current problem is, that the Entrypoint scene will get unloaded on the server and on the clients, which is a problem for the server since its not possible to spawn new players in the Entrypoint scene on the server afterwards.

To work around this issue, I have made a call to the scene manager in a script in the Entrypoint scene, which loads the Entrypoint scene and flags it as not unloaded.
The call itself is not loading any scenes, since it is already loaded, but instead just flags the Entrypoint scene to not get unloaded automatically.

Is there a more elegant solution I currently am not seeing?
Otherwise, would it be an addition to FishNet to manipulate a currently loaded scene? For example to set it should not be automatically unloaded.

```csharp
Here as an example the script that manipulates the flags for the Entrypoint scene, which itself is attach to a gameobject in said scene:
  // This call does not actually load a scene, it only flags it to be global and not automatically unloaded.
  // That is needed, because other scenes for gameplay may be loaded on the server and other clients
  // do not have the entrypoint scene anymore to connect to.
  // NOTE This is a bit fishy code, but how to solve this else?
  //      There might be no call to flag an already loaded scene, or is there?
  InstanceFinder.SceneManager.LoadGlobalScenes(new SceneLoadData(gameObject.scene)
  {
      ReplaceScenes = ReplaceOption.All,
      Options = new LoadOptions
      {
          AutomaticallyUnload = false,
          AllowStacking = false,
          LocalPhysics = LocalPhysicsMode.None
      }
  });
```

Another addition to that I am currently missing, is to get the scene load data after a scene has loaded. If it is not possible to query that data, one could of cause write a script to save that data separately, but I think there should be in general some more ways to deal with current attributes for the scene manager rather to depent all on the events and "cache"/map the data yourself.

Hope that was understandable.

Greetings,
Michiruf

**Comment by FirstGearGames** [comment]
I missed this but it sounds like you're doing it right. The flags are there to stop unloading when no clients are present. You can also laod scenes on clients only without needing to use the servers sceneManager, unless the said scene has networked objects that is.
